pkg/planner/core/task.go

```diff
@@ -881,7 +881,13 @@ func (p *PhysicalTopN) getPushedDownTopN(childPlan base.PhysicalPlan, storeTp kv
 	fixValue := fixcontrol.GetBoolWithDefault(p.SCtx().GetSessionVars().GetOptimizerFixControlMap(), fixcontrol.Fix56318, true)
 	// HeavyFunctionOptimize: if TopN's ByItems is a HeavyFunction (currently mainly for Vector Search), we will change
 	// the ByItems in order to reuse the function result.
-	if fixValue && ContainHeavyFunction(p.ByItems[0].Expr) {
+	byItemIndex := make([]int, 0)
+	for i, byItem := range p.ByItems {
+		if ContainHeavyFunction(byItem.Expr) {
+			byItemIndex = append(byItemIndex, i)
+		}
+	}
+	if fixValue && len(byItemIndex) > 0 {
```

Comment:
Could you add a plan unit test to cover this case (multiple heavy byItems and also some not-heavy byItems altogether)?